extendr-api/src/wrapper/altrep.rs

```diff
@@ -6,31 +6,49 @@ pub struct Altrep {
 }
 
 /// Rust trait for implementing ALTREP.
+/// Implement one or more of these methods to generate an Altrep class.
+/// Mechanism TBD.
 trait AltrepImpl {
-    fn unserialize_ex(_class: Robj, _state: Robj, _attr: Robj, _objf: i32, _levs: i32) -> Robj {
-        ().into()
+    /// Constructor that is called when loading an Altrep object from a file.
+    fn unserialize_ex(class: Robj, state: Robj, attributes: Robj, obj_flags: i32, levels: i32) -> Robj {
+        let res = Self::unserialize(class, state);
+        if !res.is_null() {
+            unsafe {
+                let val = res.get();
+                SET_ATTRIB(val, attributes.get());
+                SET_OBJECT(val, obj_flags);
+                SETLEVELS(val, levels);
+            }
+        }
+        res
     }
 
+    /// Simplified constructor that is called when loading an Altrep object from a file.
     fn unserialize(_class: Robj, _state: Robj) -> Robj {
         ().into()
     }
 
+    /// Fetch the state of this object when writing to a file.
     fn serialized_state(&self) -> Robj {
         ().into()
     }
 
+    /// Duplicate this object, possibly duplicating attributes.
     fn duplicate_ex(&self, _deep: bool) -> Robj {
```

Comment:
What does "ex" stand for? "extended"? Can you think of a different choice that is a little more intuitive?